lib/IR/GraphScheduler.cpp

```diff
@@ -140,6 +140,26 @@ class ChildMemSizeBasedScheduler : public Scheduler {
       orderedChildren.push_back(N->getPredicate());
     }
 
+    // SaveNode hack:
+    // We don't model memory dependencies, but we still need to honor them.
+    // Make sure the SaveNode happens after the last use of the output variable.
+    if (SaveNode *save = llvm::dyn_cast<SaveNode>(N)) {
+      Variable *output = save->getVariable();
+      for (NodeUse &use : output->getUsers()) {
+        Node *user = use.getUser();
+        if (user == save) {
+          continue;
+        }
+        // Variables may have users scattered across different functions.
```

Comment:
Sounds like it could be useful to have an iterator type that iterates only over users inside a specific function.